custom/speech.py

```diff
@@ -0,0 +1,248 @@
+else:
+    requirements = [
+        'speech_installed (install speech_recognition playsound + re-run setup.sh)']
+
+
+#global values
+num = 1
+
+
+@require(native=requirements, platform=LINUX)
+@plugin("open speech")
+def open_speech__LINUX(jarvis, s):
+    """Jarvis will open the gui for you."""
+    jarvis.say("Opening speech.......",Fore.RED)
+
+    assistant_speaks("What's your name, Human?")
+    name ='Human'
+    name = get_audio()
+    assistant_speaks("Hello, " + str(name) + '.')
+
+    while(1):
+        assistant_speaks("What can i do for you?")
+        text = str(get_audio()).lower()
+        if text == 0:
+            continue
+
+        if "exit" in str(text) or "bye" in str(text) or "sleep" in str(text):
+            assistant_speaks("Ok bye, "+ str(name)+'.')
+            break
+
+        # calling process text to process the query
+        process_text(text)
+
+
+
+
+
+# sub functions
+def assistant_speaks(output): 
+    global num 
+  
+    # num to rename every audio file  
+    # with different name to remove ambiguity 
+    num += 1
+    print("PerSon : ", output) 
+  
+    toSpeak = gtts.gTTS(text = output, lang ='en', slow = False) 
+    # saving the audio file given by google text to speech 
+    file = str(num)+".mp3"
+    toSpeak.save(file) 
+      
+    # playsound package is used to play the same file. 
+    playsound.playsound(file, True)  
+    os.remove(file) 
+
+
+
+  
+def get_audio(): 
+  
+    rObject = sr.Recognizer() 
+    audio = '' 
+  
+    with sr.Microphone() as source: 
+        print("Speak...") 
+          
+        # recording the audio using speech recognition 
+        audio = rObject.listen(source, phrase_time_limit = 5)  
+    print("Stop.") # limit 5 secs 
+  
+    try: 
+  
+        text = rObject.recognize_google(audio, language ='en-US') 
+        print("You : ", text) 
+        return text 
+  
+    except: 
+  
+        assistant_speaks("Could not understand your audio, PLease try again !") 
+        return 0
+
+
+
+def process_text(input): 
+    try: 
+        if 'search' in input or 'play' in input: 
+            # a basic web crawler using selenium 
+            search_web(input) 
+            return
+  
+        elif "who are you" in input or "define yourself" in input: 
+            speak = '''Hello, I am Jarvis. Your personal Assistant. 
+            I am here to make your life easier. You can command me to perform 
+            various tasks such as calculating sums or opening applications'''
+            assistant_speaks(speak) 
+            return
+  
+        elif "who made you" in input or "created you" in input: 
+            speak = "I have been created by Human."
+            assistant_speaks(speak) 
+            return
+  
+        elif "geeksforgeeks" in input:# just 
+            speak = """Geeks for Geeks is the Best Online Coding Platform for learning."""
+            assistant_speaks(speak) 
+            return
+  
+        elif "calculate" in input.lower(): 
```

Comment:
Doesn't this re-implement existing functionality (e.g. open, search web, ...) respectively should be a standalone plugin (e.g. wolfram search)?

Reply:
yeah, emmmm, emmmmm, emmmm, emmm, how can I say, **speech** is a plugin that can act as **jarvis teminal**, that is to say, it can execute **standard jarvis command** (like battery status), so this means, **speech** plugin can call other plugins like jarvis command **google**  or **calculate**. But the problem is that I what to get the **output** of the called plugins and say it out. If I use **jarvisAPI.eval(command)**, I can't get the output in memory, beacause all the outputs are in jarvis termimal (using print_say or jarvis.say).  My flow is: jarvis calls pluginA ---> pluginA calls pluginB --> pluginB return results to pluginA --> pluginA does something else with getting results --> pluginA can closed jarvis terminal and exit all the processing. 
So it's a little difficult for me to use google/search of jarvis, because I don't understand all the source code and jarvis are designed as independent plugins. to make things easily, I created sub-fuction search/open which I just use in speech.

emmmm, I'm currently a student and not very experienced.....maybe you can give me some guidance ^^